bot/utility/utility.py

```diff
@@ -206,6 +208,50 @@ async def welcome_message(self, user: discord.Member):
                                                               "__gemeinsam__ schaffen wir das!** :muscle:")
         await user.send(content=content, embed=embed)
 
+    @commands.command(name='format')
+    async def format(self, ctx: commands.Context, language: Optional[str], *code: str):
+        """Handler for the `format` command. Formats the provided message using clang-format and prints it using markdown syntax highlighting
+            Refer to the !howto format for an extended explanation of how to use it
+        Args:
+            language: The language abbreviation to highlight the resulting code with
+
+        """
+        try:
+            if language is None:
+                raise Exception("No language provided")
+            code = " ".join(code) if ctx.message.reference is None else ctx.message.reference.resolved.content
+            # mypy would erroneously find an error in the next two lines because it doesn't notice that at this point, `code` is always a str.
+            # You can verify its correctness by renaming the variable `code` as assignment target in the previous and all subsequent lines
```

Comment:
Have you tried doing sth. like: 

```
code: str = " ".join(code) if ctx.message.reference is None else ctx.message.reference.resolved.content
```
or alternatively
```
code = " ".join(code) if ctx.message.reference is None else str(ctx.message.reference.resolved.content)
```

That is, if mypy annoys you enough to do this.

Reply:
Both of the proposed solutions don't seem to work. I also tried with mypy --allow-redefinition, but to no avail.